src/app/machines/views/MachineList/MachineListTable/MachineListTable.tsx

```diff
@@ -846,8 +846,36 @@ export const MachineListTable = ({
     [hiddenColumns, showActions]
   );
 
+  const getMachinesDisplayed = () => {
+    if (!machineCount) {
+      return null;
+    }
+
+    const totalPages = Math.ceil(machineCount / pageSize);
+
+    if (currentPage === totalPages) {
+      return pageSize - (totalPages * pageSize - machineCount);
+    } else {
+      return pageSize;
+    }
+  };
+
   return (
     <>
+      {machineCount ? (
+        <div className="u-flex--between u-flex--align-center">
+          <strong>
+            Showing {getMachinesDisplayed()} out of {machineCount} machines
+          </strong>
```

Comment:
Have you thought about moving this to a separate component along with the shown count logic?